validphys2/src/validphys/pseudodata.py

```diff
@@ -101,8 +102,21 @@ def read_replica_pseudodata(fit, context_index, replica):
     tr["type"], val["type"] = "training", "validation"
 
     pseudodata = pd.concat((tr, val))
-    pseudodata.sort_index(level=range(1, 3), inplace=True)
 
+    # In order for this function to work also with old fit, it is necessary to remap the names
+    # being read (since the names in the context have already been remapped)
+    # The following checks whether a given name is in both the context and the fit, and if not
+    # tries to get it from the old_to_new mapping.
+    mapping = {}
+    context_datasets = context_index.get_level_values("dataset").unique()
+    for dsname in pseudodata.index.get_level_values("dataset").unique():
+        if dsname not in context_datasets:
+            new_name, _ = legacy_to_new_map(dsname)
+            mapping[dsname] = new_name
+
+    pseudodata.rename(mapping, level=1, inplace=True)
+
+    pseudodata.sort_index(level=range(1, 3), inplace=True)
     pseudodata.index = sorted_index
```

Comment:
I don't want to remove it because I'm not sure why that's there and don't want to add (more) noise to this PR.

Reply:
Well currently it would overwrite whatever the index of the psuedodata is, so unless you trust the `sorted_index` more than the one stored in the file, I'd say this is not a good idea to keep around